uart_example/lib/main.dart

```diff
@@ -0,0 +1,342 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_reactive_ble/flutter_reactive_ble.dart';
+import 'package:location_permissions/location_permissions.dart';
+
+
+// This flutter app demonstrates an usage of the flutter_reactive_ble flutter plugin
+// This app works only with BLE devices which advertise with a Nordic UART Service (NUS) UUID
+Uuid _UART_UUID = Uuid.parse("6E400001-B5A3-F393-E0A9-E50E24DCCA9E");
+Uuid _UART_RX   = Uuid.parse("6E400002-B5A3-F393-E0A9-E50E24DCCA9E");
+Uuid _UART_TX   = Uuid.parse("6E400003-B5A3-F393-E0A9-E50E24DCCA9E");
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter_reactive_ble example',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+        visualDensity: VisualDensity.adaptivePlatformDensity,
+      ),
+      home: MyHomePage(title: 'Flutter_reactive_ble UART example'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key, this.title}) : super(key: key);
+  final String title;
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final flutterReactiveBle = FlutterReactiveBle();
+  List<DiscoveredDevice> _foundBleUARTDevices = [];
+  StreamSubscription<DiscoveredDevice> _scanStream;
+  Stream<ConnectionStateUpdate> _currentConnectionStream;
+  StreamSubscription<ConnectionStateUpdate> _connection;
+  QualifiedCharacteristic _txCharacteristic;
+  QualifiedCharacteristic _rxCharacteristic;
+  Stream<List<int>> _receivedDataStream;
+  TextEditingController _dataToSendText;
+  bool _scanning = false;
+  bool _connected = false;
+  String _logTexts = "";
+  List<String> _receivedData = [];
+  int _numberOfMessagesReceived = 0;
+
+  void initState() {
+    super.initState();
+    _dataToSendText = TextEditingController();
+  }
+
+  void refreshScreen() {
+    setState(() {});
+  }
+
+  void _sendData() async {
+      await flutterReactiveBle.writeCharacteristicWithResponse(_rxCharacteristic, value: _dataToSendText.text.codeUnits);
+  }
+
+  void onNewReceivedData(List<int> data) {
+    _numberOfMessagesReceived += 1;
+    _receivedData.add( _numberOfMessagesReceived.toString() + ": " + String.fromCharCodes(data));
+    if (_receivedData.length > 5) {
+      _receivedData.removeAt(0);
+    }
+    refreshScreen();
+  }
+
+  void _disconnect() async {
+    await _connection.cancel();
+    _connected = false;
+    refreshScreen();
+  }
+
+  void _stopScan() async {
+    await _scanStream.cancel();
+    _scanning = false;
+    refreshScreen();
+  }
+
+  Future<void> showNoPermissionDialog() async {
+    return showDialog<void>(
+      context: context,
+      barrierDismissible: false, // user must tap button!
```

Comment:
I don't think this comment is useful. Yes, it explains the meaning of the parameter in words simpler than the parameter name (`barrierDismissible:`), but are you going to put such comments on every line now?